packages/postcss-lowercase-text/README.md

```diff
@@ -0,0 +1,191 @@
+# postcss-lowercase-text
+
+Postcss plugin to safely lowercase your `CSS` selectors and properties in order to minimize your gzip size
+
+## Installation
+
+```shell
+npm install postcss-lowercase-text --save
+```
+
+## Usage
+
+Refer the [PostCSS Documentation](https://github.com/postcss/postcss#usage) for using this plugin.
+
+## Example
+
+### Selector
+
+- Input
+
+```css
+A {
+  color: red;
+}
+
+UL li {
+  display : block
+}
+
+H1#heading {
+  color: red;
+}
+
+.outerClass.INNERCLASS {
+  color: red;
+}
+
+```
+
+- Output
+
+```css
+a {
+  color: red;
+}
+
+ul li {
+  display : block
+}
+
+h1#heading {
+  color: red;
+}
+
+.outerClass.INNERCLASS {
+  color: red;
+}
+
+```
+
+### Property
+
+- Input
+
+```css
+.classname {
+  COLOR: red;
```

Comment:
Can you add test for `COLOR: RED` we should lowercase them too

Reply:
we cant lowercase the value as it can be `var` or `calc` function and inside that, lowercasing is not safe

Reply:
Ok as of now I am skipping transforming any kind of function. We can do that later after discussion

Reply:
@evilebottnawi I think `postcss-value-parser` is not enough to do this operation , here is why

for example, the following css

```css
:root {
  --main-BG-color: coral;
}
.classnames {
  color: RED;
  background-color: var(--main-BG-color);
}
```

Here the parsed value using `postcss-value-parser` is

```js
ValueParser { nodes: [ { type: 'word', sourceIndex: 0, value: 'coral' } ] }
ValueParser { nodes: [ { type: 'word', sourceIndex: 0, value: 'RED' } ] }
ValueParser {
  nodes:
   [ { type: 'function',
       sourceIndex: 0,
       value: 'var',
       before: '',
       after: '',
       nodes: [ { type: 'word', sourceIndex: 4, value: '--main-BG-color' } ] } ] }
```

Now if we add the transformation method like this

```js
rule.walkDecls(decl => {
  // Transform each property declaration here
  decl.prop = decl.prop.toLowerCase();

  const parsedVal = parser(decl.value);
  decl.value = parsedVal
    .walk(node => {
      if (node.type === "function") {
        return node;
      } else {
        if (node.type === "word") {
          node.value = node.value.toLowerCase();
        }
      }
      return node;
    })
    .toString();
});
```

So this will show the result

```css
:root {
  --main-bg-color: coral;
}
.rooot {
  color: red;
  background-color: var(--main-bg-color);
}
```

Which is **Incorrect**, cause its transforming the var `main-BG-color` -> `main-bg-color`
We can skip transforming the `function` type value but in `var` , inside the variable is of type `word` so our method will transform that to lowercase.
`postcss-value-parser` doesn't support parent property in the AST so we cant confirm whether the `word` type has `function` parent or not.